### PR TITLE
Upgrade Twitter connector from V1.1 to V2

### PR DESCRIPTION
**Pull request by GihanAyesh**
## Purpose
Improvement for the issue https://github.com/wso2-enterprise/wso2-apim-internal/issues/2648

Important changes

- Endpoint security is changed from OAuth 1.0 to OAuth 2.0
- Remove endpoints that Twitter has discontinued when moving to v2
- Add new endpoints that are useful such as getMe, addDirectMessage, getListById, unlikeTweet, and getUserByUsername
- Update the documentation
- Implement new request bodies and query params
- Store both access and refresh tokens in the registry
- Implement call with retry
- New folder structure and synapse components for easy user readability

### CODE REVIEW

File: src/main/resources/twitter_config/init.xml [hunk 1]
```diff
@@ -17,29 +17,120 @@
...
+                                </enrich>
+                            </then>
+                        </filter>
+                    </then>
+                </filter>
+            </then>
+        </filter>
     </sequence>
-</template>
+</template>
```
Comment (SanojPunchihewa) [inline src/main/resources/twitter_config/init.xml]: Let's add an empty line after this. Please check other places as well

File: src/main/resources/connector.xml [hunk 2]
```diff
@@ -17,21 +17,13 @@
  ~  under the License.
 -->
 <connector>
-    <component name="twitter" package="org.wso2.carbon.connector">
+    <component name="twitter" package="org.wso2.carbon.connectors">
```
Comment (SanojPunchihewa) [inline src/main/resources/connector.xml]: Do we need to change this?

**Comment by SanojPunchihewa** [comment]
Have we tested this in the Integration Studio? Don't we need the `uischemas` to populate the views? Please check that. You can refer to the following for more information.

https://github.com/wso2-extensions/esb-connector-salesforcerest/tree/master/src/main/resources/uischema

File: src/main/resources/twitter_config/getAccessToken.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    var existingRefreshToken =mc.getConfiguration().getRegistry().getResource(refreshTokenRegistryPath);
+
+                    if(existingRefreshToken == null){
+                    mc.getConfiguration().getRegistry().newResource(refreshTokenRegistryPath,false);
+                    }
+                    mc.getConfiguration().getRegistry().updateResource(refreshTokenRegistryPath,newRefreshToken);
+                    ]]>
+                </script>
+
+                <property name="uri.var.accessToken.reg" expression="$ctx:uri.var.accessToken"/>
```
Comment (SanojPunchihewa) [inline src/main/resources/twitter_config/getAccessToken.xml]: Do we use this `uri.var.accessToken.reg` property in the rest of the mediation flow?

Comment (GihanAyesh) [reply src/main/resources/twitter_config/getAccessToken.xml]: By setting the registry value here, developer can make the next proxy server call without the access token value.

File: src/main/resources/connector.xml (hunk 2 above)
Comment (GihanAyesh) [reply src/main/resources/connector.xml]: Without that 
`ERROR {SynapseConfiguration} - Unable to deploy synapse import:{org.wso2.carbon.connectors}twitter. Required library not found.`
issue arises

Comment (SanojPunchihewa) [reply src/main/resources/connector.xml]: We usually have `org.wso2.carbon.connector` as the package in all the connectors. It seems like an issue with the import configs in `EI_HOME/repository/deployment/server/synapse-configs/default/imports/`.

**Comment by SanojPunchihewa** [comment]
We need to change the connector version to `3.0.0` in the `pom.xml`

**Review by SanojPunchihewa — APPROVED** [review]

**Pull request merged**